internal/providers/terraform/azure/dns_zone.go

```diff
@@ -0,0 +1,65 @@
+package azure
+
+import (
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetAzureRMDNSZoneRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_dns_zone",
+		RFunc: NewAzureRMDNSZone,
+		ReferenceAttributes: []string{
+			"resource_group_name",
+		},
+	}
+}
+
+func NewAzureRMDNSZone(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	group := d.References("resource_group_name")
+	location := group[0].Get("location").String()
+
+	if strings.HasPrefix(strings.ToLower(location), "usgov") {
+		location = "US Gov Zone 1"
+	}
+	if strings.HasPrefix(strings.ToLower(location), "germany") {
+		location = "DE Zone 1"
+	}
+	if strings.HasPrefix(strings.ToLower(location), "china") {
+		location = "Zone 1 (China)"
+	}
+	if location != "US Gov Zone 1" && location != "DE Zone 1" && location != "Zone 1 (China)" {
+		location = "Zone 1"
+	}
+
+	costComponents := make([]*schema.CostComponent, 0)
+
+	costComponents = append(costComponents, HostedPublicZoneCostComponent(location))
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+func HostedPublicZoneCostComponent(location string) *schema.CostComponent {
```

Comment:
The same thing with lower-case
```suggestion
func hostedPublicZoneCostComponent(location string) *schema.CostComponent {
```